src/bootstrap/compile.rs

```diff
@@ -313,12 +313,17 @@ pub fn std_cargo(builder: &Builder<'_>, target: TargetSelection, stage: u32, car
     let compiler_builtins_c_feature = if builder.config.optimized_compiler_builtins {
         if !builder.is_rust_llvm(target) {
```

Comment:
You can use `compiler-rt` without Rust patches. I think the important thing is that `compiler-rt` is built from the same sources as LLVM. So.. either `llvm-config` is unset, or llvm_from_ci is active and we're managing submodules?